osquery/tables/system/linux/pci_devices.cpp

```diff
@@ -240,6 +241,121 @@ Status PciDB::getSubsystemInfo(const std::string& vendor_id,
   return Status::success();
 }
 
+Status enrichPCISubsysInfo(Row& row,
+                           std::string& subsystem_attr,
+                           const std::string& vendor_id,
+                           const std::string& model_id,
+                           const PciDB& pcidb) {
+  boost::algorithm::to_lower(subsystem_attr);
+  std::vector<std::string> subsystem_ids;
+  boost::split(subsystem_ids, subsystem_attr, boost::is_any_of(":"));
+
+  if (subsystem_ids.size() != 2) {
+    return Status::failure(
+        "Expected 2 identifiers from sysFs device subsystem attribute, but "
+        "got " +
+        std::to_string(subsystem_ids.size()));
+  }
+
+  row.emplace("subsystem_vendor_id", "0x" + subsystem_ids[0]);
+  row.emplace("subsystem_model_id", "0x" + subsystem_ids[1]);
+
+  std::string content;
+  if (pcidb.getVendorName(subsystem_ids[0], content).ok()) {
+    row.emplace("subsystem_vendor", std::move(content));
+  }
+
+  if (pcidb
+          .getSubsystemInfo(
+              vendor_id, model_id, subsystem_ids[0], subsystem_ids[1], content)
+          .ok()) {
+    row.emplace("subsystem_model", std::move(content));
+  }
+
+  return Status::success();
+}
+
+static inline void delUdevDevice(udev_device* d) {
+  udev_device_unref(d);
+}
```

Comment:
Why do you need such simple function here?

Reply:
Good catch .  No need.